specifyweb/frontend/js_src/lib/components/QueryBuilder/Formatter.tsx

```diff
@@ -23,42 +27,73 @@ export function QueryFieldFormatter({
     () =>
       // Some code duplication, but required by TypeScript
       (type === 'formatter'
-        ? formatters?.formatters.map(({ table, name, title }) => ({
+        ? formatters?.formatters.map(({ table, name, title, isDefault }) => ({
             table,
             name,
             title,
+            isDefault,
           }))
-        : formatters?.aggregators.map(({ table, name, title }) => ({
+        : formatters?.aggregators.map(({ table, name, title, isDefault }) => ({
             table,
             name,
             title,
+            isDefault,
           }))
       )
         ?.filter(({ table }) => table?.name === tableName)
-        .map(({ name, title = name }) => ({ name, title })),
+        .map(({ name, title = name, isDefault }) => ({
+          name,
+          title,
+          isDefault,
+        })),
     [type, formatters, tableName]
   );
+
+  const [formatterSelectIsOpen, setFormatterSelect] = React.useState(false);
+
+  const id = useId('formatters-selection');
+
   return availableFormatters === undefined ? (
     typeof formatter === 'string' ? (
       <>{commonText.loading()}</>
     ) : null
   ) : availableFormatters.length > 1 ? (
     <>
-      {mappingElementDivider}
-      <div>
-        <Select
-          disabled={handleChange === undefined}
-          value={formatter}
-          onValueChange={handleChange}
-        >
-          <option />
-          {availableFormatters.map(({ name, title }, index) => (
-            <option key={index} value={name}>
-              {title}
-            </option>
-          ))}
-        </Select>
-      </div>
+      <Button.Small
+        aria-controls={id('list')}
+        aria-label={queryText.chooseFormatter()}
+        className={`${
+          availableFormatters.find((selected) => selected.name === formatter)
+            ?.isDefault ||
+          formatter === undefined ||
+          formatter === ''
+            ? 'bg-white dark:bg-neutral-600'
+            : 'bg-yellow-250 dark:bg-yellow-900 '
+        }`}
+        title={queryText.chooseFormatter()}
+        onClick={() => setFormatterSelect(!formatterSelectIsOpen)}
+      >
+        {icons.cog}
+      </Button.Small>
+      {formatterSelectIsOpen && (
+        <div>
+          <Select
+            aria-label={queryText.chooseFormatter()}
+            className={`${customSelectElementBackground}`}
+            disabled={handleChange === undefined}
+            id={id('list')}
+            value={formatter}
+            onValueChange={handleChange}
+          >
+            <option />
```

Comment:
I know the `queryField.dataObjFormatter` can be undefined, in which case the empty option makes sense. 
However, if there is a formatter/aggregator defined it appears the backend will utilize that formatter even if the field is not default and the `formatname` passed to the backend is `" "`. 
The same applies for if _all_ formatters for a table are set `default=false`: the backend will still return formatted results (I am not sure how the logic decides which to use, but it may just select the first one). 

In the below video for example, the formatter being used for CollectingEvent table is `CollectingEvent`, even when the undefined option is selected. 

https://github.com/specify/specify7/assets/64045831/f8a4b034-6b2b-4251-a18e-a074d18d4150

(There is another bug presented in the video as well. The `CollectingEvent` formatter is a default, so it theoretically should automatically be selected in the Select element, just as `Locality` is)